### ISSUE TITLE
Commands with lots of subcommands contort the help summary output

### ISSUE THREAD

**Issue by djerius** [sbakker/perl-Term-CLI#12]
I've got a command with lots of sub-commands; this turns the help summary output into a tall thin, unreadable column:

![image](https://user-images.githubusercontent.com/196875/146397634-0df71806-991b-448e-b21d-3289480d67c2.png)

**Comment by sbakker** [comment]
Ouch. That's ugly. I _thought_ I had mitigated for this already (using a maximum `=over` length, but that patch may have gotten lost. :-(
I'll have a look.

**Comment by sbakker** [comment]
Can you show me what `help --pod` spits out? The maximum `=over` width should be: (*terminal-width* - 4) / 2

**Comment by sbakker** [comment]
For example, if I add your `filter` command definition to my version (using tutorial#16 to build on), the `help` command shows me this:
```
  Commands:
    cp path path2 ...                    copy files
    debug cmd ...                        debug commands
    do action while activity             Do action while activity
    echo [arg ...]                       print arguments to stdout
    exit [excode]                        exit bssh
    filter {add|collection|copy|delete|edit|list|load|reset|save}
                                         manipulate filters
    help [cmd ...]                       show help
    interface iface {down|up}            Turn iface up or down
    ls [arg ...]                         list file(s)
    make target when                     make target at time when
    set {delimiters|verbose}             set CLI parameters
    show {clock|load|terminal}           show system properties
    sleep time                           sleep for time seconds
```

That is on a  window with 78 columns, so the maximum `=over` should be (78-4)/2 = 37, and that's what I see with `help --pod`:
```
=head2 Commands:

=over 37
```

**Comment by sbakker** [comment]
@djerius, did you ever find out why the output is so squeezed? If this still occurs with the latest git code, I'd like to see your output from `help --pod`.

**Comment by djerius** [comment]
@sbakker Sorry for the delay. 

I had to perform some surgery on the code to get down to a simple example ([test.txt](https://github.com/sbakker/perl-Term-CLI/files/7790944/test.txt)).  

I've run it against Term::CLI version 0.053006 (which looks like that's the latest in git).

If the tty width is 150 or 151, I get this:

![Screenshot from 2021-12-29 19-27-42](https://user-images.githubusercontent.com/196875/147712503-39794ba2-85e9-4c5c-8b50-01ee2c320eaf.png)

If the tty width is greater than 151, it actually sends Pod::Text::Termcap (called via Term::CLI::Command::Help::_format_pod) into what seems to be an infinite loop, constantly appending to `$output`.

**Comment by djerius** [comment]
I forgot to add that  with a tty width of 151, `help --pod` gives

```
=head2 Commands:

=over 73
```
It hangs with a size > 151.

**Comment by sbakker** [comment]
That is *so* weird. On my (Fedora 35, GNOME terminal), it works fine, both on 151 characters wide and wider. I'm almost suspecting an issue with the underlying termcap. I'd like to reproduce this. Can you give me more details about your setup, like: 

 * OS (if Linux, which distribution)
 * Terminal emulator
 * Value of `$TERM`
 * Version of Perl
 * Version of `Pod::Text::Termcap`

A most vexing issue, this! ;-)

**Comment by djerius** [comment]
I'm running 

- Debian 11 & CentOS 7.8
- Tilix (Debian), Gnome Terminal (Debian, CentOS), xterm (Debian, CentOS), rxvt (Debian)
- $TERM:
   - Tilix: xterm-256color
   - rxvt: rxvt-unicode-256color
   - xterm: xterm
   - Gnome Terminal: xterm-256color
- Perl 5.28.3
- Pod::Text::Termcap 4.10

**Comment by djerius** [comment]
Here's an even smaller code example which fails:
[term-cli-bug2.txt](https://github.com/sbakker/perl-Term-CLI/files/7803433/term-cli-bug2.txt)

The good news is that I have another installation of Perl where it works, so that I can do some more debugging

**Comment by djerius** [comment]
Aha.  Pod::Text::Termcap is shipped with Perl, and version 4.10 is shipped with Perl 5.28.3.  If I upgrade to v 4.11, the text script works!  Here's the relevant changelog entry for [podlaters 4.11](https://metacpan.org/dist/podlators/changes#L60):

>  [Pod::Text::Termcap] Correctly honor the width option, which was being
>  ignored due to a bug in interpreting user-supplied options.  Thanks,
>  eponymous alias.  (#124447)

There are fixes to Pod::Text::Termcap in subsequent releases, so requiring 4.14 (which is the current version and is two years old at the time of this writing) might be wise.

**Comment by sbakker** [comment]
Great detective work, thanks!

I just checked with my `perbrew` installation of Perl 5.14.1 and the version of `Pod::Text::Termcap` is even older than that (2.06). I decided that, instead of insisting on 4.11 or higher, it's better to check for the version of `Pod::Text::Termcap` adjust for that.

I have two options when an old version of `Pod::Text::Termcap` is detected:

 1. Limit the `=over` width (to 37).
 2. Use `Pod::Text::Overstrike`.

Using the first option, the text is sort of displayed correctly, but the "tag" line of the block can be broken as well:

```
  Commands:
    filters
    {add|collection|copy|delete|edit|list|load|reset|resolve|save}
                                         manipulate filters

    help [cmd ...]                       show help
```

Using the second option, the above text (with `=over 37`) looks like this:

```
  Commands:
    filters {add|collection|copy|delete|edit|list|load|reset|resolve|save}
                                         manipulate filters

    help [cmd ...]                       show help
```

Mind you, if it is not passed through a pager that understands overstrike sequences, the text comes out "plain" (no bold or underline). This is usually not a problem for `Term::CLI`, since it always tries to spawn a `less` pager that understands these sequences.

I slightly favour the second option. It honours the terminal width, so formatting is consistent. The only issue is that it may lose the bold/underline if and only if there is an old version of `Pod::Text::Termcap` *and* there is no `less` pager. I think I can live with that.
